R/ForecastRegrTask.R

```diff
@@ -0,0 +1,59 @@
+#' @title Create a task for univariate forecasting
+#'
+#' @rdname Task
+#' 
+#' @description Creates a task for univariate forecasting learners
+#' 
+#' @export
+makeForecastRegrTask = function(id = deparse(substitute(data)), data, target,
+  weights = NULL, blocking = NULL, frequency = 1L, fixup.data = "warn",
+  check.data = TRUE) {
+  assertString(id)
+  assertClass(data, "xts")
+  assertString(target)
+  assertInteger(frequency, lower = 0L, max.len = 1L)
+  assertChoice(fixup.data, choices = c("no", "quiet", "warn"))
+  assertFlag(check.data)
+
+  data = data.frame(row.names = index(data), coredata(data))
+
+  if (fixup.data != "no") {
+    if (is.integer(data[[target]]))
+      data[[target]] = as.double(data[[target]])
+  }
+
+  task = makeSupervisedTask("fcregr", data, target, weights, blocking, fixup.data = fixup.data, check.data = check.data)
```

Comment:
There has not been a discussion!

I chose `fcregr` because I wanted it to be short and allow for a possible future `fcclassif`. It could be changed to `forecastregr`. Though we use `regr` for regression types so I thought keeping it short would be nice